mlflow/pipelines/pipeline.py

```diff
@@ -71,8 +71,10 @@ def run(self, step: str = None) -> None:
         :return: None
         """
 
-        # Save the step requested by the user so it can be saved as a tag
-        self._initial_config_step = step
+        # Save the run parameters for later
+        self._run_args = {
+            "step": step
```

Comment:
Is there a way to automatically capture all run_args from the function? Something from inspect package could help.
I have not investigated carefully, but `inspect.getargvalues(inspect.currentframe())` may work. If it's too complicated to implement, don't bother. The current approach is not great but OK.

Reply:
@jinzhang21 
I was looking for a solution for this before and the only thing I found is https://stackoverflow.com/a/10724602/18956398
It seems to be pretty hacky so I didn't use it - but maybe it's ok? 🤔 I don't feel super confident in python, not sure what's elegant and what's not.

Reply:
The solution looks fine (not as hacky to me). But we don't have to do it in this PR. You can submit the code as is. We could discuss what to do when it comes to the point of extending the `run()` API.